angular-spa-ui/js/search-app/tags-module/tags-list-directive/tags-list-ctrl.js

```diff
@@ -0,0 +1,215 @@
+            return $scope.getArticleItem($scope.inputParam);
+        }
+    };
+
+    // Directive of module tags
+    //-------------------------
+
+    // Get list tags
+    $scope.getTagsListData = function() {
+        tagService.getTags()
+            .then(
+                // Implementation in the case of success
+                function(tagsList) {
+                    $scope.tags = tagsList.data;
+                },
+                function(errorMessage) {
+                    console.warn(errorMessage);
+                }
+            );
+    };
+
+
+    // Directive of module article
+    //----------------------------
+
+    // Get item article
+    $scope.getArticleItem = function(articleItem) {
+        console.log($scope);
+        console.log(articleItem);
+        $scope.getTagByArticleId(articleItem._id);
+    };
+
+    // Get tag by article
+    $scope.getTagByArticleId = function(articleId) {
+        tagService.getTagByArticleId(articleId)
+            .then(
+                // Implementation in the case of success
+                function(itemArticleData) {
+                    $scope.getTagsListByArticleId(itemArticleData.data);
+                },
+                function(errorMessage) {
+                    console.warn(errorMessage);
+                }
+            );
+    };
+
+    // Get tag by article
+    $scope.getTagsListByArticleId = function(listTags) {
+        _.each(listTags, function(itemTag){
+            $scope.getTagById(itemTag);
+        });
+    };
+
+    // Get list tag of one article
+    $scope.getTagById = function(itemTag) {
+        tagService.getTagById(itemTag.tagId)
+            .then(
+                // Implementation in the case of success
+                function(tagsList) {
+                    $scope.applyTagsListData(tagsList);
+                },
+                function(errorMessage) {
+                    console.warn(errorMessage);
+                }
+            );
+    };
+
+    // Remove or detach the current tag of current scope
+    //----------------------------
+
+    // Remove tag
+    $scope.removeTagItem = function(itemTag) {
+        if ($scope.currentScope === 'tagsListForTagsPage') {
+            return $scope.deleteTagItem(itemTag);
+        } else
+        if ($scope.currentScope === 'tagsListByArticleId') {
+            return $scope.detachTagFromArticle(itemTag, $scope.inputParam);
+        }
+    };
+
+    // Delete item tag
+    $scope.deleteTagItem = function(itemTag) {
+        tagService.deleteTagItem(itemTag._id)
+            .then(
+                // Implementation in the case of success
+                function() {
+                    $scope.tags = _.without($scope.tags, itemTag);
+                },
+                function(errorMessage) {
+                    console.warn(errorMessage);
+                }
+            );
+    };
+
+    // Detach item tag from article
+    $scope.detachTagFromArticle = function(itemTag, articleItem) {
```

Comment:
"Detach" normally useful for unbinding events processes, my be remove would be better